### PR TITLE
chore(inline)

### PR DESCRIPTION
**Pull request by dirvine**
Tag simple getters as inline to help compilers not set up for LTO etc.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -51,6 +51,9 @@ pub mod treemeta;
 /// This module contains TreeNode.
 pub mod treenode;
 
+/// Errors
+pub mod error;
```
Comment (dan-da) [inline src/lib.rs:55]: this appears to reference error.rs, which is not in the PR.  Does it compile?

Comment (dirvine) [reply src/lib.rs:55]: Ah missed this, will sort it now.

File: examples/tree.rs [hunk 2]
```diff
@@ -108,7 +108,7 @@ impl<ID: TreeId, TM: TreeMeta, A: Actor + std::fmt::Debug> Replica<ID, TM, A> {
 
         let mut v: Vec<&Clock<A>> = self.latest_time_by_replica.values().collect();
         v.sort_unstable_by(|a, b| a.cmp(b));
-        v.pop() 
+        v.pop()
```
Comment (dan-da) [inline examples/tree.rs:111]: this pop() has different logic (remove from end) than the original code (remove from start).

I do not see a remove-from-start method for Vec, which is why I implemented it with if/else.

Comment (dirvine) [reply examples/tree.rs:111]: Ah I see, This still passed tests so perhaps we need a test to catch this?

File: src/lib.rs (hunk 1 above)
Comment (dirvine) [reply src/lib.rs]: Perhaps we need to define the sort criteria as you had it but reverse it then pop or 

```rust
v.sort()
v.reverse()
v.pop()
```

File: examples/tree.rs (hunk 2 above)
Comment (dan-da) [reply examples/tree.rs:111]: I could do that...  a little background though:

So the "tests" in examples/tree are not formal unit/integration tests, but more "let's tryout/demonstrate this concept" that I made as I was going through the paper, to get some hands-on experience with things like log truncation, garbage collection, etc.  That's why I put them under examples, not tests.  The causal threshold calculation itself is one of these example demonstrations, ie it is not part of the library.

I'm not sure yet if causal threshold calcs should be part of this crate's API or not.  Wherever we add it, I would make more formal test(s) for it then.

anyway, the "test" that "fails" with pop() is:

```
$ ./target/debug/examples/tree test_truncate_log
generating move operations...
applying 79 operations to all 5 replicas...

truncating log of replica 15938833573310947648...
causally stable threshold: Some(Clock { actor_id: 14439148956079359196, counter: 30 })

truncating log of replica 10069411709320156414...
causally stable threshold: Some(Clock { actor_id: 14439148956079359196, counter: 30 })

truncating log of replica 14439148956079359196...
causally stable threshold: Some(Clock { actor_id: 14439148956079359196, counter: 30 })

truncating log of replica 10243984693958159930...
causally stable threshold: Some(Clock { actor_id: 14439148956079359196, counter: 30 })

truncating log of replica 11453357061113662838...
causally stable threshold: Some(Clock { actor_id: 14439148956079359196, counter: 30 })

-- Stats -- 

[
    Stat {
        replica: 15938833573310947648,
        ops_before_truncate: 79,
        ops_after_truncate: 1,
    },
    Stat {
        replica: 10069411709320156414,
        ops_before_truncate: 79,
        ops_after_truncate: 1,
    },
    Stat {
        replica: 14439148956079359196,
        ops_before_truncate: 79,
        ops_after_truncate: 1,
    },
    Stat {
        replica: 10243984693958159930,
        ops_before_truncate: 79,
        ops_after_truncate: 1,
    },
    Stat {
        replica: 11453357061113662838,
        ops_before_truncate: 79,
        ops_after_truncate: 1,
    },
]
```

Notice that ops_after_truncate() is 1 here, for all replicas.  This holds true across multiple runs.  The operations are randomly generated, so when it is working, this number changes from run to run.

tl;dr -- I can make another test with fixed input ops that gives a deterministic result that can be checked for in this case.  But really these "tests" in examples/tree.rs should not be considered formal tests (without significant refactoring).  Those are under src/ and tests/

Comment (dirvine) [reply examples/tree.rs:111]: Ah nice one, yes that is good background. Yea we should get some integration / proptest (quickcheck) in here now. The code's looking good and I feel async may even improve some parts (merging/tree traversal etc.) and simplify even more.

Comment (dan-da) [reply examples/tree.rs:111]: There are integration tests with quickcheck under /tests.
